content/articles/fastify-svelte-app/index.md

```diff
@@ -0,0 +1,466 @@
+
+- Data fetching and building components in Svelte.
+
+### Prerequisites
+
+1. Basics of programming with the [JavaScript](https://www.w3schools.com/js/DEFAULT.asp) programming and comfortability of the [Node.js](https://nodejs.org/en/) environment.
+
+2. Have [Node.js](https://nodejs.org/en/) JavaScript runtime installed on your development system.
+
+3. An IDE or code editor such as [VS Code](https://code.visualstudio.com/download).
+
+4. A basic understanding of [REST APIs](https://en.wikipedia.org/wiki/Representational_state_transfer) will be helpful.
+
+5. A web browser such as [Google Chrome]((https://www.google.com/chrome/)) on your development machine.
+
+### Setting Up The Backend with Node.js and Fastify
+
+To start the project, create a folder and name it `fastify-svelte-app`. Inside this folder, we will create another folder for our svelte client and name it `svelte-client`. Our backend code will be at the root of the main `fastify-svelte-app` folder. This is the folder we will start for now. Open the terminal and run the command `npm init -y` to initialize the npm with default configurations while creating a `package.json` file to manages our dependencies.
+
+Before initializing the Node.js server, we want to add the `fastify` and `fastify-cors` as a dependency to our project:
+
+`npm i fastify fastify-cors --save`
+
+### Setting up Fastify
+On our `index.js` file, add the following code for the server setup:
+
+```js
+
+// Require the framework and instantiate it
+const app = require('fastify')({logger: true})
+
+// handle CORS
+app.register(require('fastify-cors'), { 
+    origin: true,
+    methods: ["GET","POST", "DELETE", "PUT", "PATCH"]
+  })
+
+//Parse Content-Type
+app.addContentTypeParser('*', function (request, payload, done) {
+    var data = ''
+    payload.on('data', chunk => { data += chunk })
+    payload.on('end', () => {
+      done(null, data)
+    })
+  })  
+
+app.get('/', (req, res)=>{
+    res.send('You are in the index route')
+})
+
+app.listen(3000, (err, address)=>{
+    if(err){
+        app.log.error(err)
+        process.exit(1)
+    }
+    app.log.info(`Server listening on ${address}`)
+})
+```
+
+Here is what is happening in the above code:
+
+- First, we load the Fastify application object, instantiate it and enable logs to the console.
+
+- `app.register(require('fastify-cors')` is the method that we use to Cross-Origin Resource Sharing (CORS) in our server. This is a mechanism that restricts the request of resources on a web server depending on the domain of the HTTP request. When using the cors middleware, we can allow or limit some routes. In our case, we will allow for all routes.
+
+- Next, we declare an index route that will respond with JSON response. Our application will receive and respond with JSON as the parsed data format. 
+
+- The `app.listen()` method is the code snippet that is listening on port 3000 where the application will receive requests.
```

Comment:
It's good that you have explained well.

But, it would be great you move each point under its respective code snippet. (Break down this code snippet) By doing this, the reader would not need to move back and forth to view the code and its explanation.

Reply:
I will add descriptive comments on the code itself. If I got you right?

Reply:
Nope.

I meant something like:

Code snippet 1

Explanation 1

Code  snippet 2 

Explanation 2

Since, we break them down into smaller code snippets, make sure to add GitHub link containing full code.